include/deal.II/lac/arpack_solver.h

```diff
@@ -160,18 +160,36 @@ class ArpackSolver : public Subscriptor
   };
 
   /**
-   * Standardized data struct to pipe additional data to the solver, should it
-   * be needed.
+   * Standardized data struct to pipe additional data to the solver.
    */
   struct AdditionalData
   {
-    const unsigned int number_of_arnoldi_vectors;
-    const WhichEigenvalues eigenvalue_of_interest;
-    const bool symmetric;
-    AdditionalData(
+    /**
+     * Constructor. By default, set the number of Arnoldi vectors (Lanczos
+     * vectors if the problem is symmetric) to 15. Set the solver to find the
+     * eigenvalues of largest magnitude for a non-symmetric problem).
+     */
+    explicit AdditionalData(
       const unsigned int number_of_arnoldi_vectors = 15,
       const WhichEigenvalues eigenvalue_of_interest = largest_magnitude,
       const bool symmetric = false);
+
+    /**
+     * Number of Arnoldi/Lanczos vectors. This number should be less than the
+     * size of the problem but greater than 2 times the number of eigenvalues
```

Comment:
i think it is recommended to be around that, but strictly speaking it does not have to.

Reply:
I just looked at the code and we have an assert that checks for that value see [here](https://github.com/dealii/dealii/pull/5553/files/f6d6897b5d5873e89813632811cceddd1797c842..e82eed83b10a36fa80ce390e3e52da9f44f2feb6#diff-8fb043a1e58ccb9048d7210b151893a0R472). I have no idea if that constraint makes sense or not.

Reply:
interesting... keep it as is then